app/data/models/User.py

```diff
@@ -40,17 +48,18 @@ def validatePassword(self):
         # min 8 characters, max 20
         numberOfCharacters = r"\.{8, 20}"
 
-        return re.match(capitalLetter, password) and re.match(smallLetter, password) and re.match(oneNumber, password) and re.match(numberOfCharacters, password)
-    
-    def ifUserExists(self):
+        # return re.match(capitalLetter, password) and re.match(smallLetter, password) and re.match(oneNumber, password) and re.match(numberOfCharacters, password)
+
+        return True
 
-        if(UsersRepository.fetchUserByName(self.username)) or UsersRepository.fetchUserByEmail(self.email):
-            return True
+    def getUserDetails(self):
 
-        return False
+        return [self.id, self.username, self.email, self.password, self.birthday]
 
-    def saveUserToDataBase(self):
-        if self.ifUserExists():
-            return False
 
-        return UsersRepository.saveUser(self)
+    def ifUserExists(self):
```

Comment:
is it possible for created User object not to exist in database? isnt that obj created by UsersRepository?

Reply:
also, you are inside User class so no 'user' in method name

Reply:
I did it for validation. I create a new object of the user using the date from the registration form and then I check if that user exits already in the database.